api/document/js/schema.ts

```diff
@@ -2,15 +2,15 @@ import { Node, NodeSpec, Schema } from 'prosemirror-model';
 
 const listSchemaNodes: { [name: string]: NodeSpec } = {
   list: {
-    content: 'listitem+',
+    content: 'listitem*',
     group: 'block',
     toDOM: () => ['ol', { class: 'node-list' }, 0],
   },
   listitem: {
     attrs: {
       marker: { default: '1.' },
     },
-    content: 'block+',
+    content: 'block*',
```

Comment:
There's a section in the [ProseMirror guide](https://prosemirror.net/docs/guide/) that says:

> It is recommended to always require at least one child node in nodes that have block content (such as "doc" and "blockquote" in the example above), because browsers will completely collapse the node when it's empty, making it rather hard to edit.

Obviously our plugin will make sure that the node is deleted instead of being an empty node that the user can't see; however, my general concern is that I'm not sure if this notion of having a less stringent schema in exchange for having plugins that "fix up" the doc is something that is "doing things the ProseMirror way" or "fighting against ProseMirror".  I'm too unfamiliar with ProseMirror commands/transactions/steps in particular to be able to even answer that question, alas.

Reply:
Agreed. I get the sense that ProseMirror pushes for a strict schema with tight control over all the transforms, but thus far, that's led to a very fragile document. In that gif, for example, select-all-ing magically replaces a node with "undefined" (presumably because the underlying data model was corrupted). I think we could push this strategy further, but we'd likely need to replace many of the core transforms (including simple things, like deletion) with our own.

Taking this fix-it-later approach is definitely not my first choice, but it seems to work reasonably well. I suspect we'll find better alternatives as we progress.

Two downstream projects might be worth investigating: [Dovetail](https://github.com/heydovetail/editor/blob/4437d450a8a5164e488f8c9e0f5b8cd37bd9a442/src/schema.ts), [pubpub](https://github.com/pubpub/pubpub-packages/blob/2445714e87f25580b9c4e2c24bda837e14828cd5/packages/pubpub-editor/src/prosemirror-setup/schema/schemaDefinition.js)